chains/solana/contracts/programs/ccip-router/src/instructions/v1/onramp.rs

```diff
@@ -307,24 +305,87 @@ fn token_transfer(
 /////////////
 // Helpers //
 /////////////
-fn extra_args_or_default(default_config: Ref<Config>, extra_args: ExtraArgsInput) -> AnyExtraArgs {
-    let mut result_args = AnyExtraArgs {
-        gas_limit: default_config.default_gas_limit.to_owned(),
-        allow_out_of_order_execution: default_config.default_allow_out_of_order_execution != 0,
-    };
 
-    if let Some(gas_limit) = extra_args.gas_limit {
-        gas_limit.clone_into(&mut result_args.gas_limit)
+// process_extra_args returns serialized extraArgs, gas_limit, allow_out_of_order_execution
+// it calls the chain-specific extra args validation logic
+pub fn process_extra_args(
+    dest_config: &DestChainConfig,
+    extra_args: &[u8],
+    message_contains_tokens: bool,
+) -> Result<(Vec<u8>, u128, bool)> {
```

Comment:
Can the result be a struct instead of a tuple? It will be more easy to read and follow.